### PR TITLE
Aux/update cypress 10

### PR DESCRIPTION
**Pull request by javieri-empathy**
This is the result of all the work done to update to Cypress 12. Each commit has been already reviewed and approved.

### CODE REVIEW

**Review by tiborux — DISMISSED** [review]

File: packages/x-components/.cypress-cucumber-preprocessorrc.json [hunk 1]
```diff
@@ -1,4 +1,3 @@
 {
-  "nonGlobalStepDefinitions": true,
-  "stepDefinitions": "tests/e2e"
+  "stepDefinitions": "tests/e2e/**/*.ts"
```
Comment (LuisMartinez15) [inline packages/x-components/.cypress-cucumber-preprocessorrc.json:2]: **nitpick**: increase `stepDefinitions` regex specificity (same as in the component `specPattern` config)

```suggestion
  "stepDefinitions": "tests/e2e/**/*.spec.ts"
```

File: packages/x-components/cypress.config.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        })
+      );
+      return config;
+    },
+    specPattern: 'tests/e2e/**/*.feature'
+  },
+  component: {
+    specPattern: 'tests/unit/**/*.spec.ts',
+    supportFile: 'tests/support/index.ts',
+    indexHtmlFile: 'tests/support/component-index.html',
```
Comment (LuisMartinez15) [inline packages/x-components/cypress.config.ts:35]: **nitpick**: this is the default config value, could we remove it here? ([docs reference](https://docs.cypress.io/guides/component-testing/component-framework-configuration#Custom-Index-File)).

```suggestion
```

File: packages/x-components/package.json [hunk 3]
```diff
@@ -53,14 +53,17 @@
...
     "test:unit-watch": "jest --watch",
-    "test:e2e": "vue-cli-service test:e2e",
+    "test:e2e": "vue-cli-service test:e2e --e2e --browser chrome",
+    "test:e2e:firefox": "vue-cli-service test:e2e --e2e --browser firefox",
     "test:e2e:ci": "vue-cli-service test:e2e --headless --browser chrome",
     "test": "npm run test:unit && npm run test:e2e:ci",
-    "cypress:open": "cypress open",
-    "cypress:open:component": "cypress open-ct"
+    "cypress:open": "cypress open --e2e --browser chrome",
+    "cypress:open:firefox": "cypress open--e2e --browser firefox",
```
Comment (LuisMartinez15) [inline packages/x-components/package.json:64]: **nitpick**: relying on the implicit default behaviour of the `cypress open` command

```suggestion
    "cypress:open:firefox": "cypress open --browser firefox",
```

**typo(blocking)**: there's a missing whitespace between open and `--e2e` parameter

```suggestion
    "cypress:open:firefox": "cypress open --e2e --browser firefox",
```

File: packages/x-components/package.json [hunk 4]
```diff
@@ -53,14 +53,17 @@
     "prepublishOnly": "npm run build",
     "test:unit": "jest --silent && npm run test:unit-cypress",
     "test:unit-jest": "jest",
-    "test:unit-cypress": "cypress run-ct",
+    "test:unit-cypress": "cypress run --component",
     "test:unit-coverage": "jest --coverage",
     "test:unit-watch": "jest --watch",
-    "test:e2e": "vue-cli-service test:e2e",
+    "test:e2e": "vue-cli-service test:e2e --e2e --browser chrome",
+    "test:e2e:firefox": "vue-cli-service test:e2e --e2e --browser firefox",
```
Comment (LuisMartinez15) [inline packages/x-components/package.json:60]: **nitpick**: we are using the `--e2e` parameter to be exhaustive and explicit, but it is not necessary as it is the default behaviour.

```suggestion
    "test:e2e": "vue-cli-service test:e2e --browser chrome",
    "test:e2e:firefox": "vue-cli-service test:e2e --browser firefox",
```

**Review by LuisMartinez15 — CHANGES_REQUESTED** [review]
**issue(blocking)**: we need to solve merge conflicts in both globals `package.json` and `package-lock.json`

**praise**: thanks a lot for your hard work, hope this update improves test execution speed both locally and on the ci!

File: packages/x-components/package.json (hunk 4 above)
Comment (javieri-empathy) [reply packages/x-components/package.json:60]: I don't think it is the default value. If I run cypress without that flag I see this screen, allowing me to choose the kind of tests to run:
image

File: packages/x-components/cypress.config.ts (hunk 2 above)
Comment (javieri-empathy) [reply packages/x-components/cypress.config.ts:35]: This is not the default one, the default is: `cypress/support/component-index.html`
https://docs.cypress.io/guides/component-testing/component-framework-configuration#Custom-Dev-Server

Comment (LuisMartinez15) [reply packages/x-components/cypress.config.ts:35]: you're right, my bad!

File: packages/x-components/package.json (hunk 4 above)
Comment (LuisMartinez15) [reply packages/x-components/package.json:60]: So they lie in their own [documentation](https://docs.cypress.io/guides/guides/command-line#Options-1) 😞 

image

**Review by LuisMartinez15 — APPROVED** [review]

**Pull request merged**